x-pack/platform/plugins/private/upgrade_assistant/server/lib/rollup_job.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ElasticsearchClient } from '@kbn/core/server';
+import {
+  RollupGetRollupIndexCapsResponse,
+  RollupGetJobsResponse,
+} from '@elastic/elasticsearch/lib/api/types';
+
+export async function getRollupJobByIndexName(esClient: ElasticsearchClient, index: string) {
+  let rollupCaps: RollupGetRollupIndexCapsResponse;
+
+  try {
+    rollupCaps = await esClient.rollup.getRollupIndexCaps({ index });
```

Comment:
would something like this work? we would surface the not found case, but the function will throw if any other error happens preventing it from being swallowed 

```suggestion
    rollupCaps = await esClient.rollup.getRollupIndexCaps({ index }, { ignore: [404] });
    
    if (!rollupCaps) {
      throw new Error(`Failed to retrieve rollup caps for ${index}`);
    }
```